File: app/src/main/java/com/mmartin/ghibliapi/AppComponent.java
package com.mmartin.ghibliapi;

import com.mmartin.ghibliapi.data.GhibliFilmsRepositoryModule;
import com.mmartin.ghibliapi.network.NetworkModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;

/**
 * Created by mmartin on 10/11/17.
 */
@Singleton
@Component(modules = {ApplicationModule.class, NetworkModule.class, GhibliFilmsRepositoryModule.class})
public interface AppComponent {
    void inject(App app);

    @Component.Builder
    interface Builder {
        @BindsInstance
        AppComponent.Builder application(App application);

        AppComponent build();
    }
}

File: app/src/main/java/com/mmartin/ghibliapi/di/component/AppComponent.kt
package com.mmartin.ghibliapi.di.component

import com.mmartin.ghibliapi.App
import com.mmartin.ghibliapi.di.module.ApplicationModule
import dagger.Component

/**
 * Created by mmartin on 10/11/17.
 */
@Component(modules = arrayOf(ApplicationModule::class))
interface AppComponent {
    fun getApp(): App
}
